src/node_messaging.cc

```diff
@@ -103,22 +115,25 @@ MaybeLocal<Value> Message::Deserialize(Environment* env,
   EscapableHandleScope handle_scope(env->isolate());
   Context::Scope context_scope(context);
 
-  // Create all necessary MessagePort handles.
-  std::vector<MessagePort*> ports(message_ports_.size());
-  for (uint32_t i = 0; i < message_ports_.size(); ++i) {
-    ports[i] = MessagePort::New(env,
-                                context,
-                                std::move(message_ports_[i]));
-    if (ports[i] == nullptr) {
-      for (MessagePort* port : ports) {
-        // This will eventually release the MessagePort object itself.
-        if (port != nullptr)
-          port->Close();
+  // Create all necessary objects for transferables, e.g. MessagePort handles.
+  std::vector<BaseObject*> host_objects(transferables_.size());
+  for (uint32_t i = 0; i < transferables_.size(); ++i) {
```

Comment:
use `foreach` would be better?

Reply:
I don't think `foreach` makes things simpler: `i` is required as the index for `transferables_` and `host_objects`.